core/src/retransmit_stage.rs

```diff
@@ -319,28 +323,35 @@ fn retransmit_shred(
         cluster_nodes.get_retransmit_addrs(slot_leader, key, root_bank, data_plane_fanout)?;
     let addrs: Vec<_> = addrs
         .into_iter()
-        .filter(|addr| ContactInfo::is_valid_address(addr, socket_addr_space))
+        .filter(|addr| socket_addr_space.check(addr))
         .collect();
     compute_turbine_peers.stop();
     stats
         .compute_turbine_peers_total
         .fetch_add(compute_turbine_peers.as_us(), Ordering::Relaxed);
 
     let mut retransmit_time = Measure::start("retransmit_to");
-    let num_nodes = match multi_target_send(socket, shred, &addrs) {
-        Ok(()) => addrs.len(),
-        Err(SendPktsError::IoError(ioerr, num_failed)) => {
-            stats
-                .num_addrs_failed
-                .fetch_add(num_failed, Ordering::Relaxed);
-            error!(
-                "retransmit_to multi_target_send error: {:?}, {}/{} packets failed",
-                ioerr,
-                num_failed,
-                addrs.len(),
-            );
-            addrs.len() - num_failed
-        }
+    let num_nodes = match cluster_nodes::get_broadcast_protocol(key) {
+        Protocol::QUIC => addrs
+            .iter()
+            .map(|addr| quic_connection_cache.get_connection(addr).send_data(shred))
+            .filter(Result::is_ok)
```

Comment:
separately, you can just do `.filter_map(|addr| quic_connection_cache.get_connection(addr).send_data(shred))`, right?